castai/resource_node_template.go

```diff
@@ -909,7 +908,7 @@ func flattenCustomTaints(taints *[]sdk.NodetemplatesV1Taint) []map[string]string
 			t[FieldValue] = toString(v)
 		}
 		if e := taint.Effect; e != nil {
-			t[FieldEffect] = toString(e)
+			t[FieldEffect] = string(*e)
```

Comment:
why changing this? The toString is guarantee no nil pointer

Reply:
`toString` accepts `*string` but the type of `taint.Effect` is `*NodetemplatesV1TaintEffect` which even though is alias cannot be passed as `*string` without casting. There's a check `e != nil` so it's not going to be nil. I can make it `toString((*string)(e))` if you think it it's better?